canonical-web-and-design/vanilla-framework
# Update Vanilla docs to use fixed width layout with side navigation

## Done

Updated Vanilla navigation to use recommended documentation layout:

- responsive side navigation
- dark header
- updated existing content to fit new layout
- added some custom JS to scroll side navigation active item into view and to prevent scrolling document when mobile side navigation drawer is open

Fixes #2974 

## QA

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/)
- [Browse new docs](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/), check if everything looks OK
  - try different docs pages ([code](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/base/code), etc)
  - some use grid: ([icons](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/patterns/icons), [component status](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/component-status)
  - check updated [examples listing](https://vanilla-framework-canonical-web-and-design-pr-2988.run.demo.haus/docs/examples)
  - try search
  - check home page and other pages

Make sure to try on different screen sizes (with side navigation visible, and with a toggle on small screens). 

- open side navigation drawer on small screen
  - verify that side nav can be scrolled, but document body doesn't scroll
  - resize screen back to desktop size, make sure scroll works as expected


## Screenshots

<img width="1152" alt="Screenshot 2020-04-20 at 13 55 19" src="https://user-images.githubusercontent.com/83575/79749000-ab9a0500-830e-11ea-9b7d-ea84d5740199.png">

## Tasks

- [x] what to do with hero on docs home page? - removed
- [x] use dark navigation
- [x] codepens are even smaller that previously
- [x] update pages that use grid in content
- [x] what to do with table of contents (currently removed)
  - experiment with adding to side navigation
  - DONE, but it seems quite useless for pages very low in side navigation, because you need to scroll to table of contents anyway
- [x] sticky side navigation
- [x] make active side nav element scroll into view
- [x] disable scroll on body when mobile side navigation is open (via JS)
